packages/block-library/src/gallery/shared.js

```diff
@@ -21,3 +21,19 @@ export const pickRelevantMediaFiles = ( image, sizeSlug = 'large' ) => {
 	}
 	return imageProps;
 };
+
+/**
+ * The new gallery block format is not compatible with the use_BalanceTags option
+ * in WP versions <= 5.8 https://core.trac.wordpress.org/ticket/54130. The
+ * window.wp.galleryBlockV2Enabled flag is set in lib/compat.php. This method
+ * can be removed when minimum supported WP version >=5.9.
+ */
+export function isGalleryV2Enabled() {
+	// We want to fail early here, at least during beta testing phase, to ensure
+	// there aren't instances where undefined values cause false negatives.
+	if ( ! window.wp || typeof window.wp.galleryBlockV2Enabled !== 'boolean' ) {
```

Comment:
Just wondering, if this PR is backported to the WP core beta, then will the logic in `gutenberg_check_gallery_block_v2_compatibility` also be backported? Without it, in 5.9, with Gutenberg deactivated, where `wp.galleryBlockV2Enabled` isn't set, will this throw an error?

In that case, it could be better to treat the v2 gallery block setting being `undefined` as `true` 🤔. I might be missing some context with the impact on mobile, though!

Reply:
Good question, Andrew. For the mobile perspective, I'm going to test making `true` default (because some users will not have a cached value when the editor is initialized):

```diff
diff --git a/packages/edit-post/src/editor.native.js b/packages/edit-post/src/editor.native.js
index b045724a04..7f71d7f68a 100644
--- a/packages/edit-post/src/editor.native.js
+++ b/packages/edit-post/src/editor.native.js
@@ -31,7 +31,9 @@ class Editor extends Component {
 		super( ...arguments );
 
 		// need to set this globally to avoid race with deprecations
-		window.wp.galleryBlockV2Enabled = props.galleryWithImageBlocks;
+		// defaulting to true to avoid issues with a not-yet-cached value
+		const { galleryWithImageBlocks = true } = props;
+		window.wp.galleryBlockV2Enabled = galleryWithImageBlocks;
 
 		if ( props.initialHtmlModeEnabled && props.mode === 'visual' ) {
 			// enable html mode if the initial mode the parent wants it but we're not already in it
```

For mobile, we won't face the same backporting issue, since users on older versions won't encounter this logic, and users on newer versions will default to `true` (assuming all goes well with testing :point_up_2: ). I wonder if something similar would work (or be appropriate) for web? Or perhaps that is an error that we'd like to surface in beta (if it exists)?

Reply:
Thanks Matthew, nice to see that's how we'll handle the situation in mobile 👍. Let's see what @glendaviesnz thinks for the web when he's back on tomorrow 🙂

Reply:
@andrewserong `gutenberg_check_gallery_block_v2_compatibility` will also need to be pulled into WP 5.9. The explicit error we are throwing here is mostly for test purposes to see if we ever hit that explicit exception during testing - I wasn't thinking we would leave that in in there permanently, but instead more likely default to `true` once we are comfortable there aren't any funny edge cases we have overlooked that would have that value as `undefined` when it really should be `false` - does that make sense?

Reply:
Yes, thanks for confirming @glendaviesnz! That does sound like a good way to handle it for the beta and so that issues don't go unnoticed.

It sounds like in that case we might want to open up a PR in [wordpress-develop](https://github.com/wordpress/wordpress-develop) to add in the `check_gallery_block_v2_compatibility` function prior to this PR landing, since I _think_ the PHP code gets merged in separately from the JS being released in npm packages for 5.9?

Reply:
@andrewserong `check_gallery_block_v2_compatibility` is in the gutenberg repo and [added as part of this PR](https://github.com/WordPress/gutenberg/pull/36191/files#diff-83705774c15f7e3df5b54b2096fe476721e8b962ba319a8e7dc46d7b9b0b27b4R207), so was thinking that it gets included in 5.9 when this PR merge is cherry picked? I don't see any of the other methods in `lib/compat.php anywhere in the `wordpress-develop` repo, am I missing something?

Reply:
@glendaviesnz good question — I'm not 100% sure of the order in which things go into 5.9. I believe there are two steps: commits for 5.9 are cherry picked into the [wp/trunk](https://github.com/wordpress/gutenberg/tree/wp/trunk) branch and then new npm package releases are published (e.g. [in this recent commit](https://github.com/WordPress/gutenberg/commit/ca22e8852fcc91889510e417bdaa180ad60f8dac)). Then, the next step is that over in `wordpress-develop` there's a PR created that updates trunk to point to the new packages (e.g. https://github.com/WordPress/wordpress-develop/pull/1883).

It looks like that latter PR includes some PHP changes in addition to bumping the package numbers. I'm not sure if these changes needed to be made manually? So for the case of `check_gallery_block_v2_compatibility`, it'd be good to confirm where it'll be merged (or how best to do it) before we merge this PR.

CC: @noisysocks — just a question for the 5.9 release, this PR includes an additional function `check_gallery_block_v2_compatibility` that currently lives in `lib/compat.php` in Gutenberg, but will need to exist in core. Do we need to open up a separate PR in `wordpress-develop` for this function, or will it get rolled in when the PR is cherry picked into 5.9?

Reply:
When I prepare the Core patch that updates packages (e.g. https://github.com/WordPress/wordpress-develop/pull/1883) every Monday, I do a diff of what PHP has changed in the Gutenberg plugin (`git diff $old_sha wp/trunk **/*.php`)  and then manually make those changes in Core.

For big PHP changes, I appreciate if the PR author can open a ticket in Core Trac and make those changes themselves. That makes it less likely I have to stay up late on Monday 😀

For smaller changes such as this one, it's okay to leave it with me. You can help me (or any future release lead) out by leaving a note in the doc comment which describes how the change should look once added to Core.